nextpnr.dockerfile

```diff
@@ -29,10 +28,9 @@ RUN mkdir -p /tmp/nextpnr/build \
 FROM build AS build-ecp5
 COPY --from=hdlc/pkg:prjtrellis /prjtrellis /
 
-RUN mkdir -p /tmp/nextpnr/build \
- && cd /tmp/nextpnr \
- && curl -fsSL https://codeload.github.com/YosysHQ/nextpnr/tar.gz/master | tar xzf - --strip-components=1 \
- && cd build \
+RUN git clone --depth 1 https://github.com/YosysHQ/nextpnr.git /tmp/nextpnr \
```

Comment:
There is an experimental feature that allows using resources from previous stages without increasing the size of the current stage. See:

- https://github.com/ghdl/docker/blob/master/vunit.dockerfile#L35
- https://github.com/ghdl/docker/blob/master/vunit.dockerfile#L1

That would allow you to clone the repo once only.

Anyway, this is not critical.

Reply:
What about splitting the stages here, and base each build on that same layer, where the git is fetched once?

Reply:
Yes, that makes sense. Cloning the repo twice was an overlook!

At the same time, `build-all` should be `build-generic`. There is no need for building `nextpnr-ice40` and `nextpnr-ecp5` twice! See https://github.com/hdl/containers/pull/10#discussion_r557044028.

However, if you want to contribute that, better do it in a PR for that only. That will allow us to merge it faster, and #10 might be rebased on top of it.

Reply:
This:
> At the same time, build-all should be build-generic
and
> See [#10 (comment)](https://github.com/hdl/containers/pull/10#discussion_r557044028).

resolves some confusion with building some parts of `nextpnr` twice
and I will integrate that.